app/views/shared/_block_link.html.erb

```diff
@@ -2,10 +2,32 @@
 yields: Link text (required).
 locals:
 * url: URL of link.
+* new_tab: Whether link should open in a new tab. Defaults to false. Use best judgment to reserve
+           new tabs to when absolutely necessary, such as when form data may otherwise be lost.
 %>
-<% content = yield.presence or raise "no block content given" %>
-<%= tag.a href: url, class: 'usa-link block-link' do %>
+<%
+content = yield.presence or raise "no block content given"
+new_tab = local_assigns.fetch(:new_tab, false)
+tag_attrs = { href: url, class: 'usa-link block-link' }
+tag_attrs[:target] = '_blank' if new_tab
+%>
+<%= tag.a tag_attrs do %>
   <%= content %>
+  <% if new_tab %>
+    <svg
+      xmlns="http://www.w3.org/2000/svg"
+      height="10"
+      viewBox="0 0 64 55"
+      focusable="false"
+      aria-hidden="true"
+    >
+      <path
+        d="M35.226 5c.333 0 .605.106.818.32s.319.485.319.816v2.273c0 .332-.107.604-.32.817s-.484.32-.817.32H10.227c-1.562 0-2.9.555-4.013 1.668s-1.669 2.45-1.669 4.013v29.545c0 1.563.556 2.9 1.669 4.013s2.45 1.668 4.013 1.668h29.546c1.562 0 2.9-.555 4.013-1.668a5.47 5.47 0 0 0 1.668-4.013V33.41c0-.332.106-.604.32-.817s.484-.32.816-.32h2.273c.331 0 .604.107.817.32s.32.485.32.818v11.363C50 47.591 49 50 46.999 52s-4.41 3.001-7.226 3.001H10.227C7.41 55 5.001 54 3 51.999s-3-4.41-3-7.226V15.227c0-2.817 1-5.226 3-7.226S7.41 5 10.227 5zm26.46-5c.627 0 1.17.229 1.627.686s.687 1 .687 1.626v18.501c0 .626-.229 1.168-.687 1.626a2.22 2.22 0 0 1-1.626.687c-.626 0-1.168-.23-1.626-.687l-6.36-6.36-23.558 23.56c-.24.24-.518.36-.831.36s-.59-.12-.831-.36l-4.12-4.12a1.142 1.142 0 0 1-.361-.831c0-.313.12-.59.361-.83l23.559-23.56-6.36-6.36c-.457-.457-.686-1-.686-1.626s.229-1.168.687-1.626 1-.686 1.625-.686z"
+        fill="currentColor"
+      />
+    </svg>
```

Comment:
Rather than rendering this SVG inline, we should probably just include the `usa-link--external` class to the wrapping link, to be sure the icon styling is consistent and to automatically fix the line breaking behavior.

The downside is that we don't get automatic text color inheriting via `currentColor`, as in case of a visited link:

![image](https://user-images.githubusercontent.com/1779930/130142064-8c14457f-df1b-486a-b91b-426cce2f0b51.png)

Reply:
I know USWDS has been using [CSS mask](https://developer.mozilla.org/en-US/docs/Web/CSS/mask) as a workaround for coloring icons in other similar situations (e.g. the "official USA website" banner arrow icon), so maybe an opportunity for an upstream enhancement.